### DISCUSSION TITLE
minian for low SNR recordings made in ventral brain regions through thin relay lenses?

### DISCUSSION THREAD

**Discussion by Gabrielle-Siemonsmeier** [miniscope/minian#192] [Q&A]
Hi everyone,

I'm looking to transition from MIN1PIPE to minian to process my miniscope recordings, but I'm having a hard time getting the pipeline to work for me.

For context, I'm using V3 UCLA miniscopes to image ventral striatum (-4.6mm DV, from Bregma) in mice using 0.6 x 7.3mm relay GRIN lenses. I've imaged both CaMKIIa-GCaMP6f in WT mice and CAG-FLEX-GCaMP7s in D1-Cre mice. I've been trying to run homecage recordings (2-3min in length at 15-20fps) from several of my animals, which I had successfully processed with MIN1PIPE previously, through minian, without success so far (see attached pictures below of max projection after preprocessing and motion correction: 216, 235, 237 are WT, 289, 292, 295 are D1-Cre; sorry for how crude these are, can share other visualizations if useful).

<img width="524" alt="216_comparison" src="https://user-images.githubusercontent.com/82478676/157921641-2f0afd39-13db-4bc3-9073-221e53116f88.png">
<img width="524" alt="235_comparison" src="https://user-images.githubusercontent.com/82478676/157921645-e68cecb7-23ba-485c-b0be-127f6be51d67.png">
<img width="524" alt="237_comparison" src="https://user-images.githubusercontent.com/82478676/157921647-2fd643ef-930b-47a0-87cd-dfb5a4341188.png">
<img width="524" alt="289_comparison" src="https://user-images.githubusercontent.com/82478676/157921649-80011421-7bba-4e27-ba54-fd4983b1421a.png">
<img width="524" alt="292_comparison" src="https://user-images.githubusercontent.com/82478676/157921651-23f735c6-7414-43ef-9d75-bfc980965e40.png">
<img width="524" alt="295_comparison" src="https://user-images.githubusercontent.com/82478676/157921654-005d7fc6-1942-4b9f-9ab4-31e4b394602c.png">

By contrast, a 50s recording made of a Cre-dependent GCaMP6f in the dorsal striatum of an A2a-Cre mouse through a 1mm diameter relay lens was processed beautifully in both pipelines (see attached picture below of max projection after preprocessing and motion correction: 3249). I reckon this recording more closely resembles the demo data, which I understood to be dorsal CA1, presumably imaged through a single 1.8mm GRIN lens (correct me if I'm wrong!)

<img width="633" alt="3249_comparison" src="https://user-images.githubusercontent.com/82478676/157921659-0836148d-a88c-4c98-898e-22ab4c5eec7a.png">

Here's a dropbox folder containing the final minian visualization videos: https://www.dropbox.com/sh/ite4qp76xkp8ad9/AACwTt0mQi7UzDU8052DNo_Qa?dl=0 (let me know if there's any access issues!)
You can tell from the top right that motion correction is often unsatisfactory (and motion artifacts get 'baked in'), and from the bottom left and right that cell identification and trace extraction seems to fail, sometimes entirely.

The main differences between my dorsal vs. ventral striatum recordings seem to me to be the following:
- the FOV in my ventral striatum recordings covers a smaller area with a higher magnification, and cells therefore appear larger, with a much smaller number of cells in total
- the SNR is lower, with lower levels of fluorescence overall (recordings seem 'dim', with fluorescence sometimes not or barely visible to the naked eye)
- activity of each cell is sparser, with some cells showing only one or two peaks of fluorescence over the course of the entire recording
- sometimes, but not always, ventral striatum recordings have much higher levels of motion

In MIN1PIPE, I've used sr=0.5 (spatial downsampling by a factor of 2) and se=5 or 6 (structural element for background removal) for the dorsal striatum recording (as well as sr=1 and se=10 or 12, which yielded a number of units closer to what I obtained with minian, but a projection and neural contours that looked less accurate), and sr=0.25 (spatial downsampling by a factor of 4) and se=10 (structural element for background removal) for the ventral striatum recordings. I stuck quite closely to the default minian parameters for the dorsal striatum recording, but I haven't found a set of parameters in minian that would approximate what I obtain in MIN1PIPE for the ventral striatum recordings (but maybe I'm testing the completely wrong ranges of values?), and I have had to skip the background removal step altogether.

Bottom line, I was wondering if anyone had successfully used minian for recordings from brain regions other than dorsal CA1, and in particular, any ventral brain regions that would require thin relay lenses; or more generally, recordings with low SNR, sparse cellular activity, and/or high levels of motion. Any insight, advice, tips or tricks appreciated!

Cheers,

Gabrielle

**Comment by phildong** [comment]
I personally don't have experience in this brain region, but I could throw in some ideas based on the info you sent:
1. you don't necessarily need to spatial-downsample. (if you do you have to change parameters accordingly). In any case the window size you need in `remove_background` should probably be much larger than the default. (This parameter should basically correspond to your `se` parameter in MIN1PIPE). You mentioned you have to skip this step altogether, I'm curious why and I'd say this is the parameter you have to play with.
2. sometimes when the cell activities are too sparse the motion correction can have a hardtime capturing the correct shift. See my answer in #194  for how to potentially improve mc. But I'd say for initial debugging purpose you could try skipping mc for now as long as your data don't have huge amount of shift.

**Reply by Gabrielle-Siemonsmeier** [reply]
Hi Phil,
Thanks for getting back to me!

1. I haven't spatially downsampled in minian, actually! I did in min1pipe which is why I mentioned it alongside the se parameter I used then. Wouldn't se in min1pipe correspond to ksize for the denoising step in minian rather than wnd for the background removal step?
Regarding the latter, I have skipped that step because it seemingly removes both background and signal in my videos (though to be honest I had the same impression with the glow removal step; could that one be skipped without having too much of a negative impact on subsequent processing? ETA: nevermind, just tried that for mouse 237 and the denoising looks completely off without prior glow removal), and creates odd-looking circular artefacts; a coworker doing recordings in PFC told me she'd decided to skip background removal for the same reason. I've attached screenshots below. I've also added screenshots of all steps in the pipeline for all mice, as well as of the background removal step with wnd ranging from 10 to 45, to the [dropbox folder](https://www.dropbox.com/sh/ite4qp76xkp8ad9/AACwTt0mQi7UzDU8052DNo_Qa?dl=0) I linked above.

<img width="457" alt="backgroundremoval_wnd20" src="https://user-images.githubusercontent.com/82478676/163857435-0a935768-7d08-4e07-9259-387fc6eb331c.png">

<img width="458" alt="3backgroundremoval_wnd20" src="https://user-images.githubusercontent.com/82478676/163857542-501cf9d3-fd3e-42ec-923f-d15aa31a883c.png">

<img width="456" alt="3backgroundremoval_wnd20" src="https://user-images.githubusercontent.com/82478676/163857567-4bac1974-95eb-4a9b-af23-9fdf81eed82b.png">

<img width="459" alt="3backgroundremoval_wnd20" src="https://user-images.githubusercontent.com/82478676/163857597-a3792e74-2004-4f83-a849-e58f093895e8.png">

<img width="454" alt="3backgroundremoval_wnd20" src="https://user-images.githubusercontent.com/82478676/163857616-ee135763-0c4e-451b-8cb2-f32904d4c43d.png">

<img width="464" alt="3backgroundremoval" src="https://user-images.githubusercontent.com/82478676/163857642-e54e5dc5-3253-4b22-87c5-823be62779ec.png">

(this last one is mouse 3249 in the dorsal striatum, which as I mentioned above worked out well)


2. Unfortunately some of my videos do have quite a bit of motion. I'll try the stable landmark ROI trick, thanks!

**Reply by Gabrielle-Siemonsmeier** [reply]
just a quick update in case anyone is having similar issues: I now tried using a Cre-dependent GCaMP8s in the same D1-Cre line as well as the V4 miniscope, and am having much more luck with the processing.

<img width="241" alt="6_487_EWL40_k7_w15" src="https://user-images.githubusercontent.com/82478676/207163535-16491a3c-409c-4001-9ff7-b6231f1c91ae.png"> <img width="241" alt="487_sr0pt5se6" src="https://user-images.githubusercontent.com/82478676/207164045-98679d4d-7a54-4c8a-bf63-9294521c7396.jpg">

this is with ksize 7 and wnd 15 (minian) and sr 0.5 and se 6 (min1pipe). still working on tweaking the parameters in minian's CNMFE implementation (for now, some bona fide cells are being discarded) but it looks quite good overall.

as an aside, the ROI selection to create a mask for motion correction has never actually worked for me (as in, it doesn't seem the mask is actually created or used downstream, though I never dug much more into the issue).

**Reply by einsteinstudent005** [reply]
Hi Gabrielle, 

Do you dilute your GCaMP8 before injecting? These images are beautiful!

Thanks,
Alexa
